test-app.gradle

```
@@ -21,29 +22,54 @@ ext.applyTestAppSettings = { DefaultSettings defaultSettings ->
 ext.applyTestAppModule = { Project project, String packageName ->
     applyNativeModulesAppBuildGradle(project, packageName)
 
+    def manifestFile = file("$rootDir/app.json")
+    def manifest = new JsonSlurper().parseText(manifestFile.text)
+
+    def resources = manifest["resources"]
+
+    def androidResources = null
+    if (resources instanceof List) {
+        androidResources = resources
+    } else if (resources.containsKey("android")) {
+        androidResources = resources["android"]
+    } else {
+        throw new IllegalArgumentException("The \"resources\" property of the app.json has " +
+            "to be either a list of paths, or must contain a nested list with the \"android\" key")
+    }
+
+    def androidResourceFiles = androidResources
+        .collect { file("$rootDir/$it") }
+
+    def androidResDir = androidResourceFiles
+        .find { it.isDirectory() && it.path.endsWith("res") }
```

Should we check that the last component of the path is `res`? I.e.
```suggestion
        .find { it.isDirectory() && it.name == "res" }
```